src/index.js

```diff
@@ -43,25 +50,34 @@ export default function (CustomElement, opts) {
     static get displayName() {
       return displayName;
     }
-    componentDidMount() {
-      this.componentWillReceiveProps(this.props);
+    constructor(props) {
+      super(props);
+      this.onRef = this.onRef.bind(this);
     }
     componentWillReceiveProps(props) {
       const node = ReactDOM.findDOMNode(this);
+      this.applyProps(node, props);
+    }
+    applyProps(node, props) {
       Object.keys(props).forEach(name => {
         if (name === 'children' || name === 'style') {
           return;
         }
-
         if (name.indexOf('on') === 0 && name[2] === name[2].toUpperCase()) {
           syncEvent(node, name.substring(2), props[name]);
         } else {
           node[name] = props[name];
         }
       });
     }
+    onRef(node) {
+      if (node != null) {
```

Comment:
Oh wait, I was conflating this with Skate. React does call ref with null (but I think it's always `null`). You could probably just use `if (node)`. If you want to be 100% safe, `if (node instanceof HTMLElement)` would work, but I think that's overkill and truthy / falsy checks here work just fine.